app/assets/v2/js/activity.js

```diff
@@ -17,6 +17,74 @@ $(document).ready(function() {
     });
   }
 
+  var get_jitsi_api_object = function(roomName){
+    var jitsi_domain = 'meet.jit.si';
+    var jitsi_options = {
+      roomName: roomName,
+      parentNode: document.querySelector('#' + roomName),
+      welcomePageEnabled: true
+    }
+    var jitsi_api = new JitsiMeetExternalAPI(jitsi_domain, jitsi_options);
+    return jitsi_api;
+  }
+
+  // join video call
+  $(document).on('click', '.click_here_to_join_video', function(e) {
+    e.preventDefault();
+    if (!document.contxt.github_handle) {
+      _alert('Please login first.', 'error', 1000);
+      return;
+    }
+    if (typeof document.jitsi_api != 'undefined') {
+      _alert('You can only be in one video call at a time.', 'error', 1000);
+      return;
+    }
+    $(this).addClass('live');
+    $(this).text('');
+    var roomName = $(this).data('roomname');
+    var api = get_jitsi_api_object(roomName);
+    var avatarURL = 'https://gitcoin.co/dynamic/avatar/' + document.contxt.github_handle;
+    document.jitsi_api = api;
+    api.executeCommand('displayName', document.contxt.github_handle);
+    api.executeCommand('avatarUrl', avatarURL);
+    api.executeCommand('toggleAudio'); // default off
+    api.executeCommand('toggleVideo'); // default off
+    var participants_count = api.getNumberOfParticipants() + 1
+    var $target = $(this).parents('.activity_detail_content');
+    $target.append("<a href=# class='float-right leave_video_call'>Leave Video Call</a>")
+  });
+
+  //refresh job for live call
+  setInterval(function(){
```

Comment:
If I get it right this pull will run for everyone on the call, that can be a problem. Can we do the POST just with the room owner?

Reply:
no because the room owner may leave!  

the calls are actually pretty performant becasue the backend is only making a redis write..  no DB writes involved.

ideally one day when jitsi releasees a python API we can query it in the backend for all active calls instead of leveraging the clients JS.